shanoir-ng-studies/src/test/java/org/shanoir/ng/study/service/StudyServiceTest.java

```diff
@@ -183,24 +188,95 @@ public void updateTest() throws AccessDeniedException, EntityNotFoundException,
 	public void updateStudyUsersTest() throws EntityNotFoundException, MicroServiceCommunicationException {
 		Study existing = createStudy();
 		existing.setStudyUserList(new ArrayList<StudyUser>());
-		existing.getStudyUserList().add(createStudyUsers(1L, 1L, existing, StudyUserRight.CAN_SEE_ALL, StudyUserRight.CAN_IMPORT));
-		existing.getStudyUserList().add(createStudyUsers(2L, 2L, existing, StudyUserRight.CAN_ADMINISTRATE));
+		existing.getStudyUserList().add(createStudyUsers(1L, 1L, existing, true, StudyUserRight.CAN_SEE_ALL, StudyUserRight.CAN_IMPORT));
+		existing.getStudyUserList().add(createStudyUsers(2L, 2L, existing, true, StudyUserRight.CAN_ADMINISTRATE));
 		
 		Study updated = createStudy();
 		updated.setStudyUserList(new ArrayList<StudyUser>());
-		updated.getStudyUserList().add(createStudyUsers(1L, 1L, updated, StudyUserRight.CAN_DOWNLOAD));
-		updated.getStudyUserList().add(createStudyUsers(null, 3L, updated, StudyUserRight.CAN_SEE_ALL));
+		updated.getStudyUserList().add(createStudyUsers(1L, 1L, updated, true, StudyUserRight.CAN_DOWNLOAD));
+		updated.getStudyUserList().add(createStudyUsers(null, 3L, updated, true, StudyUserRight.CAN_SEE_ALL));
 		
 		given(studyRepository.findOne(STUDY_ID)).willReturn(existing);
 		given(studyUserRepository.findOne(1L)).willReturn(existing.getStudyUserList().get(0));
 		given(studyUserRepository.findOne(2L)).willReturn(existing.getStudyUserList().get(1));
 		List<StudyUser> in = new ArrayList<>(); in.add(updated.getStudyUserList().get(1));
-		List<StudyUser> out = new ArrayList<>(); out.add(createStudyUsers(4L, 3L, updated, StudyUserRight.CAN_SEE_ALL));
+		List<StudyUser> out = new ArrayList<>(); out.add(createStudyUsers(4L, 3L, updated, true, StudyUserRight.CAN_SEE_ALL));
 		given(studyUserRepository.save(in)).willReturn(out);
 
 		studyService.update(updated);
 	}
 
+	@Test
+	@WithMockKeycloakUser(id = 3, username = "jlouis", authorities = { "ROLE_EXPERT" })
+	public void testUpdateStudyUsersNoDUA() {
+		Study existing = createStudy();
+		existing.setStudyUserList(new ArrayList<StudyUser>());
+		existing.getStudyUserList().add(createStudyUsers(1L, 1L, existing, false, StudyUserRight.CAN_SEE_ALL, StudyUserRight.CAN_IMPORT));
```

Comment:
this should be true here

Reply:
In this testing I am testing with no DUA:
So I create a user with "false" set  in 'confirmed' field and then verify that it has been updated to true at the end